pkg/api/globalquery/client/client.go

```diff
@@ -59,3 +90,50 @@ func (c *Client) Query(ctx context.Context, args *query.Args) (*results.Result,
 
 	return res, nil
 }
+
+// Query performs the global query and returns its result, while consuming updates to partial results
+func (sse *SSEClient) Query(ctx context.Context, args *query.Args) (*results.Result, error) {
+	logger := logging.FromContext(ctx)
+
+	if sse.onFinish == nil || sse.onUpdate == nil {
+		return nil, errors.New("no event callbacks provided (onUpdate, onFinish)")
+	}
+
+	// use a copy of the arguments, since some fields are modified by the client
+	queryArgs := *args
```

Comment:
Are there any pointer sub-structs in there? If so, beware that those are copied by reference still...

Reply:
Checked, didn't seem like it...